core/src/main/java/org/elasticsearch/search/internal/InternalSearchHit.java

```diff
@@ -125,15 +125,6 @@ public int docId() {
         return this.docId;
     }
 
-    public void shardTarget(SearchShardTarget shardTarget) {
-        this.shard = shardTarget;
-        if (innerHits != null) {
-            for (InternalSearchHits searchHits : innerHits.values()) {
-                searchHits.shardTarget(shardTarget);
-            }
-        }
```

Comment:
@martijnvg can you check this out please? It seems like this innerHits if condition was never executed in prod code (this method was only called from a test). If this is copied to the remaining shard setter, `search.inner_hits/10_basic/Parent/child inner hits` fails as the index is also present in the inner_hits. I think we are ok to remove this but I wonder why it was there in the first place.